src/firebolt_db/firebolt_async_dialect.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+from types import ModuleType
+from typing import Any, Iterator, List, Optional, Tuple
+
+import firebolt.async_db as async_dbapi
+from firebolt.async_db import Connection
+from sqlalchemy.engine import AdaptedConnection  # type: ignore[attr-defined]
```

Comment:
Why is this ignore needed?

Reply:
Looks like `sqlalchemy-stubs` don't cover newer features available in `sqlalchemy.engine` so mypy thorws an error that AdaptedConnection does not exist in engine, while it clearly does. For now it's type-ignored. This error does not exist in another stub package available for sqlalchemy (which introduces some other errors), so we can potentially migrate to that one. Will need to investigate further, outside of this change.

Reply:
Ok, got you. Please leave a short comment